### PR TITLE
merge of the stable branch into master

### PR DESCRIPTION
**Pull request by ostcar**
Pull request to show differences. To not merge.

### CODE REVIEW

File: openslides/core/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
```
Comment (normanjaeckel) [inline openslides/core/exceptions.py]: Remove this line, please.

File: openslides/core/views.py [hunk 2]
```diff
@@ -211,3 +213,81 @@ class CustomSlideDeleteView(CustomSlideViewMixin, utils_views.DeleteView):
...
+        elif name.startswith('delete-tag-'):
+            try:
+                self.get_tag_queryset(name, 11).delete()
+            except TagException as error:
+                self.error = str(error)
+            except Tag.DoesNotExist:
+                self.error = 'Tag does not exist'
+            else:
+                self.action = 'deleted'
+        return super(TagListView, self).ajax_get(request, *args, **kwargs)
```
Comment (normanjaeckel) [inline openslides/core/views.py]: Here you can use `return super().ajax_get(request, *args, **kwargs)`

File: openslides/core/views.py [hunk 3]
```diff
@@ -211,3 +213,81 @@ class CustomSlideDeleteView(CustomSlideViewMixin, utils_views.DeleteView):
...
+        Also sets self.pk to the pk inside the name.
+        """
+        try:
+            self.pk = int(name[place_in_str:])
+        except ValueError:
+            raise TagException('Invalid name in request')
+        return Tag.objects.filter(pk=self.pk)
+
+    def get_ajax_context(self, **context):
+        return super(TagListView, self).get_ajax_context(
```
Comment (normanjaeckel) [inline openslides/core/views.py]: Here you can use `return super().get_ajax_cont(`

File: openslides/motion/views.py [hunk 4]
```diff
@@ -227,7 +236,53 @@ def manipulate_object(self, form):
...
+
+class MotionCreateAmendmentView(MotionCreateView):
+    """
+    Create an amendment.
+    """
+
+    def dispatch(self, *args, **kwargs):
+        if not config['motion_amendments_enabled']:
+            raise Http404('Amendments are disabled in the config.')
+        return super(MotionCreateAmendmentView, self).dispatch(*args, **kwargs)
```
Comment (normanjaeckel) [inline openslides/motion/views.py]: Here you can use `return super().dispatch(*args, **kwargs)`

File: openslides/motion/views.py [hunk 5]
```diff
@@ -227,7 +236,53 @@ def manipulate_object(self, form):
...
+        super(MotionCreateAmendmentView, self).manipulate_object(form)
+
+    def get_initial(self):
+        """
+        Sets the initial values to the form.
+
+        This are the values for title, text and reason which are set to the
+        values from the parent motion.
+        """
+        initial = super(MotionCreateAmendmentView, self).get_initial()
```
Comment (normanjaeckel) [inline openslides/motion/views.py]: Here you can use `initial = super().get_initial()`

File: tests/motion/test_views.py [hunk 6]
```diff
@@ -1,11 +1,14 @@
 import os
 import tempfile
+from mock import MagicMock
```
Comment (normanjaeckel) [inline tests/motion/test_views.py]: Here we use unittest.mock, don't we?

File: openslides/core/models.py [hunk 7]
```diff
@@ -44,3 +44,20 @@ def get_absolute_url(self, link='update'):
...
+
+    name = models.CharField(max_length=255, unique=True,
+                            verbose_name=ugettext_lazy('Tag'))
+
+    class Meta:
+        ordering = ['name']
+        permissions = (
+            ('can_manage_tags', ugettext_noop('Can manage tags')), )
+
+    def __unicode__(self):
```
Comment (normanjaeckel) [inline openslides/core/models.py]: This should be `__str__`.

**Comment by normanjaeckel** [comment]
Some remarks concerning changes to Python 3.

File: openslides/core/exceptions.py (hunk 1 above)
Comment (ostcar) [inline openslides/core/exceptions.py]: done

File: openslides/core/views.py (hunk 2 above)
Comment (ostcar) [inline openslides/core/views.py]: done

File: openslides/core/views.py (hunk 3 above)
Comment (ostcar) [inline openslides/core/views.py]: done

File: openslides/motion/views.py (hunk 4 above)
Comment (ostcar) [inline openslides/motion/views.py]: done

File: openslides/motion/views.py (hunk 5 above)
Comment (ostcar) [inline openslides/motion/views.py]: done

File: tests/motion/test_views.py (hunk 6 above)
Comment (ostcar) [inline tests/motion/test_views.py]: done

File: openslides/core/models.py (hunk 7 above)
Comment (ostcar) [inline openslides/core/models.py]: done

**Comment by normanjaeckel** [comment]
Ok.

**Pull request merged**